simObj.py

```diff
@@ -1,52 +1,58 @@
 from action import Action
 from cellCodes import CellCodes
 
+
 class ISimObj:
+    # is keeping the last action necessary?
+    # def __init__(self) -> None:
+    #     self.last_action = None
+    #
+    # def SetLastAction(self, action: Action) -> None:
+    #     self.last_action = action
 
-    def __init__(self) -> None:
-        self.last_action = None
+    def GetCode(self, x: int, y: int) -> int:
+        raise NotImplementedError
 
-    def SetLastAction(self, action: Action) -> None:
-        self.last_action = action;
 
+class Empty(ISimObj):
     def GetCode(self, x: int, y: int) -> int:
-        raise NotImplementedError
+        return CellCodes.Empty
```

Comment:
Thats dangerous, it may cause some object to pretend their cell is empty and other stuff will break. Imo better and safer to throw